New package: sqlitestudio-3.3.3

ar-jan opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

<!-- Note: If the build is likely to take more than 2 hours, please [skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration)
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->
<!-- 
#### Local build testing
- I built this PR locally for my native architecture, (ARCH-LIBC)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl
  - armv7l
  - armv6l-musl
-->

#### Benefits

[SQLiteStudio](https://sqlitestudio.pl/) makes it easy to quickly open and inspect a database while navigating your filesystem, my primary use case (it supports much more functionality).

![sqlitestudio_full_43](https://user-images.githubusercontent.com/1176293/154484792-f1637eda-5fc8-4e43-b921-d68016064bf5.png)

jackloomen commented:
Any update on the status of this PR?

ar-jan commented:
In my opinion it's ready, I'm actively using it.

paper42 commented on common/shlibs:
```diff
@@ -353,6 +353,9 @@ libpoppler-qt5.so.1 poppler-qt5-0.31.0_1
 libtcl8.6.so tcl-8.6.0_1
 libtk8.6.so tk-8.6.1_3
 libsqlite3.so.0 sqlite-3.38.0_1
+libsqlite3.so.0 sqlite-3.31.1_3
```
why? this shlib is already on the line above this one

paper42 commented on common/shlibs:
```diff
@@ -353,6 +353,9 @@ libpoppler-qt5.so.1 poppler-qt5-0.31.0_1
 libtk8.6.so tk-8.6.1_3
 libsqlite3.so.0 sqlite-3.38.0_1
+libsqlite3.so.0 sqlite-3.31.1_3
+libcoreSQLiteStudio.so.1 sqlitestudio-3.3.3_1
+libguiSQLiteStudio.so.1 sqlitestudio-3.3.3_1
```
and this is not required

paper42 commented on srcpkgs/sqlitestudio/files/sqlitestudio.desktop:
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Type=Application
+Name=SQLiteStudio
+Comment=Database manager to create, edit, and browse SQLite databases
+Icon=/usr/share/icons/hicolor/scalable/apps/sqlitestudio.svg
+Exec=sqlitestudio %f
+Terminal=false
+Categories=Database;Development;Utility;
+MimeType=application/vnd.sqlite3;application/x-sqlite3;
```
Could you open a PR upstream to include this .desktop file in their repository? It would be nice if every distribution didn't have to ship their own.

ar-jan replied:
I looked at doing that, but they have a desktop entry contained within an [XML file](https://github.com/pawelsalawa/sqlitestudio/blob/master/SQLiteStudio-installer.xml) which they're using for installation, so it looked unlikely to be accepted. You think I should still give it a try there?

paper42 commented on srcpkgs/sqlitestudio/files/sqlitestudio.desktop:
```diff
@@ -0,0 +1,9 @@
+Type=Application
+Name=SQLiteStudio
+Comment=Database manager to create, edit, and browse SQLite databases
+Icon=/usr/share/icons/hicolor/scalable/apps/sqlitestudio.svg
```
```suggestion
Icon=sqlitestudio
```
I think this could work

paper42 commented on srcpkgs/sqlitestudio/template:
```diff
@@ -0,0 +1,49 @@
+archs="~aarch* ~armv*"
+build_style=qmake
+hostmakedepends="qt5-qmake qt5-host-tools"
+makedepends="qt5-devel
+ qt5-script-devel
+ qt5-svg-devel
+ qt5-tools-devel
+ readline-devel
+ sqlite-devel
+ tcl-devel"
```
put all of them to one line and wrap at 80 columns

paper42 commented on srcpkgs/sqlitestudio/template:
```diff
@@ -0,0 +1,49 @@
+ readline-devel
+ sqlite-devel
+ tcl-devel"
+depends="sqlite"
```
this is automatically detected from shlibs

paper42 commented on srcpkgs/sqlitestudio/template:
```diff
@@ -0,0 +1,49 @@
+pkgname=sqlitestudio
+version=3.3.3
+revision=1
+archs="~aarch* ~armv*"
```
why is this necessary? If it fails to cross-compile, this is wrong and we should instead fix it (or set $nocross)

paper42 commented on srcpkgs/sqlitestudio/template:
```diff
@@ -0,0 +1,49 @@
+checksum=f234e1ea87d27e80a1794f86d32116ebd8919d51be63d35a7d0816a77115ce31
+
+do_configure() {
+	mkdir -p ${wrksrc}/output/build/Plugins
```
you are already in $wrksrc, no need to specify it explicitly

paper42 commented on srcpkgs/sqlitestudio/template:
```diff
@@ -0,0 +1,49 @@
+}
+
+do_build() {
+	cd ${wrksrc}/output/build
+	make
```
use make -C

ar-jan commented:
Thanks, I addressed your comments except the .desktop entry and the cross build. Set to nocross for now.

I gave fixing cross build another try but didn't get it working. There are many uses of `INCLUDEPATH=` and `LIBS=` in sqlitestudio's .pro and .pri files, and I guess that means those files would have to be modified instead of just adding some includes to CXXFLAGS and LDFLAGS. Is that right?

ar-jan commented:
bump
